src/extras/esm.js

```diff
@@ -0,0 +1,31 @@
+/*
+ * Interop for ESM modules to return the module export
+ */
+(function (global) {
+  var esmPtrn = /\besm?\b/;
+  function getPath(url) {
+    var parsed = new URL(url, 'http://domain');
+    return parsed.pathname + parsed.search + parsed.hash;
+  };
+  var systemJSPrototype = global.System.constructor.prototype;
+  var originalInstantiate = systemJSPrototype.instantiate;
+  systemJSPrototype.instantiate = function (url) {
+    if (!esmPtrn.test(getPath(url))) {
+      return originalInstantiate.apply(this, arguments);
+    }
+
+    return import(url).then(function (exported) {
+      return [
+        [],
+        function (_export) {
+          return {
+            execute() {
+              _export({ __useDefault: !!exported.default, ...exported });
```

Comment:
`__useDefault` should not be set for an ES module as that breaks the semantics. I'd suggest having it be `_export(exported)` only (if that works).

Reply:
I see, thanks! I wasn't sure if I needed it but I added it when saw it was used in the [`use-default` plugin](https://github.com/systemjs/systemjs/blob/master/src/extras/use-default.js#L11).